semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/SarifExtension.scala

```diff
@@ -0,0 +1,66 @@
+package io.shiftleft.semanticcpg.language
+
+import io.shiftleft.codepropertygraph.generated.Cpg
+import io.shiftleft.codepropertygraph.generated.help.Doc
+import io.shiftleft.codepropertygraph.generated.nodes.Finding
+import io.shiftleft.semanticcpg.sarif.SarifConfig.SarifVersion
+import io.shiftleft.semanticcpg.sarif.SarifSchema.{Sarif, Sarif2_1_0}
+import io.shiftleft.semanticcpg.sarif.{SarifConfig, SarifSchema, v2_1_0}
+import org.json4s.Formats
+import org.json4s.native.Serialization.{write, writePretty}
+
+import java.net.URI
+
+/** Converts findings written to the CPG to the SARIF format.
+  *
+  * @param traversal
+  *   the findings
+  * @see
+  *   https://docs.oasis-open.org/sarif/sarif/v2.1.0/sarif-v2.1.0.html
+  */
+class SarifExtension(val traversal: Iterator[Finding]) extends AnyVal {
+
+  @Doc(info = "execute this traversal and convert findings to SARIF format")
+  def toSarif(implicit config: SarifConfig = SarifConfig()): Sarif = {
+
+    def generateSarif(results: List[SarifSchema.Result] = Nil, baseUri: Option[URI] = None): Sarif = {
```

Comment:
The default parameter values are not really necessary here right? We have two call sites for this method. For clarity we can just spell out the arguments.